cli/pkg/modelregistry/client.go

```diff
@@ -286,6 +539,28 @@ func ToMetadataValueMap(
 	return props, nil
 }
 
+// FromMetadataValueMap converts a map of MetadataValue to a map of string to interface {}
+func FromMetadataValueMap(
+	metaData map[string]openapi.MetadataValue,
+) (map[string]interface{}, error) {
+	props := make(map[string]interface{})
+	for k, v := range metaData {
+		switch {
+		case v.MetadataStringValue != nil:
+			props[k] = v.MetadataStringValue.GetStringValue()
+		case v.MetadataStructValue != nil:
+			var sl StringList
+			if err := decodeFromBase64(&sl, v.MetadataStructValue.GetStructValue()); err != nil {
+				return nil, fmt.Errorf("failed to decode metadata value for %s: %w", k, err)
+			}
+			props[k] = sl.Items
```

Comment:
Correct and that's by design as the []interface{} is coming from the yaml decoding, which can't determine what's the type of a json array. And since we only support an [] of strings, so we only store `[] string` in MR, and return it as `[] string` for ease of later use.